### PR TITLE
deps(scaladoc): update flexmark from 0.42.12 to 0.62.2

### PR DESCRIPTION
**Pull request by Dedelweiss**
This pr updates the flexmark dependencies used in Scaladoc from 0.42.12, which is from 2019, up to the latest release of 0.64.0 to 0.62.2. This is mainly done to tackle a bunch of CVEs that are attached to the old versions of
flexmark.

fixes https://github.com/lampepfl/dotty/issues/16223

### CODE REVIEW

File: dist/bin/scaladoc [hunk 1]
```diff
@@ -124,6 +136,10 @@ done
 
 classpathArgs
 
+echo "----------"
+echo $jvm_cp_args | grep tables
+echo "----------"
```
Comment (ckipp01) [inline dist/bin/scaladoc:141]: Btw make sure to remove this.

File: project/Build.scala [hunk 2]
```diff
@@ -1130,6 +1130,7 @@ object Build {
     enablePlugins(DottyJSPlugin).
     dependsOn(`scala3-library-bootstrappedJS`).
     settings(
+      bspEnabled := false,
```
Comment (ckipp01) [inline project/Build.scala:1133]: Make sure to revert this.

File: project/build.sbt [hunk 3]
```diff
@@ -1,7 +1,4 @@
 // Used by VersionUtil to get gitHash and commitDate
 libraryDependencies += "org.eclipse.jgit" % "org.eclipse.jgit" % "4.11.0.201803080745-r"
 
-
-Compile / unmanagedSourceDirectories +=
-  baseDirectory.value / "../language-server/src/dotty/tools/languageserver/config"
```
Comment (ckipp01) [inline project/build.sbt:6]: Did you mean to remove this?

**Review by ckipp01 — COMMENTED** [review]
Btw you have some unrelated changes in here. Make sure to rebase on the latest main and that might fix them.

File: project/build.sbt (hunk 3 above)
Comment (Dedelweiss) [reply project/build.sbt:6]: I think this change comes from @tgodzik's commit.

File: scaladoc/src/dotty/tools/scaladoc/tasty/comments/markdown/SectionRenderingExtension.scala [hunk 4]
```diff
@@ -30,18 +36,19 @@ object SectionRenderingExtension extends HtmlRenderer.HtmlRendererExtension:
...
       val anchor = AnchorLink(s"#$id")
-      val attributes = Attributes()
       val headerClass: String = header.getLevel match
         case 1 => "h500"
         case 2 => "h500"
         case 3 => "h400"
         case 4 => "h300"
         case _ => "h50"
-      attributes.addValue(AttributeImpl.of("class", headerClass))
+      //val a: Attribute = AttributeImpl.of("class", headerClass)
```
Comment (ckipp01) [inline scaladoc/src/dotty/tools/scaladoc/tasty/comments/markdown/SectionRenderingExtension.scala:45]: Can we remove this comment?

**Review by ckipp01 — CHANGES_REQUESTED** [review]
Great job on this @Dedelweiss! Just a single nit about a comment that we can remove. Other than that this should be good to go!

**Review by ckipp01 — APPROVED** [review]
Great work getting this across the finish line @Dedelweiss. LGTM!

**Pull request merged**